staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -472,80 +478,65 @@ type processorListener struct {
 
 func newProcessListener(handler ResourceEventHandler, requestedResyncPeriod, resyncPeriod time.Duration, now time.Time) *processorListener {
 	ret := &processorListener{
-		pendingNotifications:  []interface{}{},
 		nextCh:                make(chan interface{}),
+		addCh:                 make(chan interface{}),
```

Comment:
1. Previous code acquires the lock and then adds the notification to the slice. New code pushes it via a channel. Both have potential to block.
2. The whole purpose of the `pop` method running concurrently is to receive from `addCh` as quickly as possible (or push to `nextCh`).

Benchmark shows that it's not a big deal. I think doing buffering in the channel and then re-buffering into a slice is unnecessary.